### PR TITLE
Devfile Commands support

### LINKED ISSUE #21030: Support Devfile commands in Che-Code
### Is your task related to a problem? Please describe

As a user, I want to be able to run my Devfile commands in Che-Code.
The commands should be visible as the Code's Tasks.

### Describe the solution you'd like

Before starting Che-Code, we can parse a flattened Devfile to read the commands and pass them to the `tasks.json` file.

### Describe alternatives you've considered

_No response_

### Additional context

This issue is a part of the epic https://github.com/eclipse/che/issues/20500

### Release Notes Text

Che-Code translates commands described in a Devfile as Visual Studio Code tasks. That makes it easier for developers to run predefined commands from within the IDE.

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

The PR adds the Che-Commands extension. It detects the Commands defined in a Devfile and represents them as VS Code Tasks. The environment variables substitution is supported for the `exec.workingDir` property.

The PR closes https://github.com/eclipse/che/issues/21030

#### How to test
1. Spin up a DevWorkspace using the following link:
https://che-dogfooding.apps.che-dev.x6e0.p1.openshiftapps.com/#https://github.com/azatsarynnyy/java-spring-petclinic/tree/devfilev2
2. In the command palette type a `task ` shortcut and choose a `che` task type.
3. Run `prepare-database` command. It's running in the `mariadb` component. See the output.
4. Run `maven-build` command. It's running in the `tools` component. See the output.

che-code-task

### CODE REVIEW

**Review by vzhukovs — APPROVED** [review]

**Comment by benoitf** [comment]
Code looks good but I'm not able to test on the dogfooding instance (not related to this PR)

File: code/extensions/che-commands/src/taskProvider.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		const kind: CheTaskDefinition = {
+			type: 'che',
+			command,
+			workdir,
+			component
+		};
+
+		command = command + ` --component ${component}`;
+		const execution = new vscode.ShellExecution(command, { cwd: expandEnvVariables(workdir) });
+		const task = new vscode.Task(kind, vscode.TaskScope.Workspace, name, 'eclipse-che.commands', execution);
```
Comment (RomanNikitenko) [inline code/extensions/che-commands/src/taskProvider.ts]: probably it makes sense to use a short type for `che` commands
but up to you
Screenshot 2022-06-02 at 17 56 02

Comment (RomanNikitenko) [inline code/extensions/che-commands/src/taskProvider.ts]: probably it makes sense to use a short type for `che` commands
but up to you
Screenshot 2022-06-02 at 17 56 02

Comment (azatsarynnyy) [reply code/extensions/che-commands/src/taskProvider.ts]: Good point.
Actually, it's not a task type, but a task source. I don't know what this string affects except this list. The documentation just says `Is presented in the user interface`.
Usually, an extension's publisher and name are used there, exactly how it's defined in the extension's package.json.
I've shortened it to `che`. Just pushed. Let's see how it looks and works, once the PR image is ready.

**Comment by RomanNikitenko** [comment]
Is it expected to have an ability to edit `che` task configurations at the current step of an implementation?

I'm asking because I tried to do it and found that it's impossible to execute a configured `che` task.
So, I applied `Configure Task` action and changed just a value for the `command` field.
I get `There is no task provider registered for task of type che` notification at running the task.


https://user-images.githubusercontent.com/5676062/171665878-97f44a0d-432e-4f83-8b8b-1289ec9c9613.mov


Screenshot 2022-06-02 at 18 40 00

**Comment by azatsarynnyy** [comment]
> Is it expected to have an ability to edit che task configurations at the current step of an implementation?

No. Currently, just running a task detected in the user's Devfile.

File: code/extensions/che-commands/src/taskProvider.ts (hunk 1 above)
Comment (azatsarynnyy) [reply code/extensions/che-commands/src/taskProvider.ts]: Now, it looks much better:
image

**Review by RomanNikitenko — APPROVED** [review]

**Comment by azatsarynnyy** [comment]
Thank you @RomanNikitenko for your review and testing.

**Pull request merged**